aten/src/ATen/native/mps/OperationUtils.mm

```diff
@@ -917,6 +917,22 @@ void executeMPSAllocatorCallback(void* ptr, EventType event) override {}
   return std::make_shared<MetalKernelFunction>(cpl, func);
 }
 
+MetalKernelFunction* MetalShaderLibrary::getCachedKernelFunctionPtr(const std::string& name) {
+  // Check if kernel is already cached
+  auto it = kernelCache.find(name);
```

Comment:
Do we have to use another cache here?
1. getLibraryPipelineState already uses a cache;
2. The way I suggested was to return `new MetalKernelFunction(cpl, func)` directly, and then manage that MetalKernelFunction object lifetime in the generated code in a RAII way. You can look at how AtenRecordFunctionHandle and RAIIAtenRecordFunctionHandle work as a reference. As a matter of fact, I suspect you don't need to create this function, you only need to create a new object in `aoti_torch_mps_get_kernel_function`.

Reply:
I am not sure about the implementation details about MTLFunction and MTLLibrary, but I am ok if want to use another cache here. One thing I will recommend is to rename `getCachedKernelFunctionPtr` to `borrowCachedKernelFunctionPtr`, so that the caller knows they don't own the returned pointer and thus not manage its lifetime.